packages/vite-plugin-svelte/src/index.js

```diff
@@ -67,14 +63,31 @@ export function svelte(inlineOptions) {
 				log.debug('additional vite config', extraViteConfig, 'config');
 				return extraViteConfig;
 			},
-			// @ts-ignore Allow exist in vite 6
-			configEnvironment(name, config) {
+
+			// @ts-ignore This hook only works in Vite 6
+			async configEnvironment(name, config, opts) {
 				config.resolve ??= {};
+
+				// Emulate Vite default fallback for `resolve.mainFields` if not set
+				if (config.resolve.mainFields == null) {
+					// These exports only exist in Vite 6
+					const { defaultClientMainFields, defaultServerMainFields } = await import('vite');
```

Comment:
this or `import * as vite from 'vite'` ?

Reply:
I'd lean on this for now I think so when/if we only support Vite 6, we get cleaner named imports

Reply:
as discussed on discord we could also just bump vite to 6 and release a major

Reply:
i'd really prefer a new major of vps for this, otherwise we have to extend the test matrix to cover vite5 and 6 too

Reply:
So should we start the prerelease on `main` now or a different branch?

Reply:
update deps on main first and release the pending release even though its mostly cosmetic so that 4.0.1 goes out from main. after that we have to branch of v4 like we did for the previous ones and then start pre on main

Reply:
`main` is now in pre mode